interface/controller/item_controller.go

```diff
@@ -0,0 +1,54 @@
+package controller
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"academy_bootcamp/application/usecase"
+	"academy_bootcamp/domain/model"
+)
+
+type itemController struct {
+	Usecase usecase.ItemService
+}
+
+type ItemController interface {
+	GetItemById(c Context) error
+	// CreateItem(c Context) error
+}
+
+func NewItemController(us usecase.ItemService) ItemController {
+	return &itemController{us}
+}
+
+func (uc *itemController) GetItemById(c Context) error {
+	var item *model.Item
+	key, err := strconv.ParseUint(c.Param("id"), 10, 16)
+	if err != nil {
+		return errors.New("QueryParam invalido")
+	}
+
+	item, err2 := uc.Usecase.GetByKey(key)
+
+	if err2 != nil {
+		return err2
+	}
+	return c.JSON(http.StatusOK, item)
+}
```

Comment:
Normally, err assignation and checks are performed one after the other with no empty line.

Also, you should declare variables as soon as you are going to use them; if there's an error in line 27, there was no sense having created the `item` var in line 26. 

Even better, there's no need to declare it beforehand; you could use the short variable declaration. Go will infer its type, and this way we could also reuse `err`, instead of creating new errors here and there.

```suggestion
	key, err := strconv.ParseUint(c.Param("id"), 10, 16)
	if err != nil {
		return errors.New("QueryParam invalido")
	}
	
	item, err := uc.Usecase.GetByKey(key)
	if err != nil {
		return err
	}
	
	return c.JSON(http.StatusOK, item)
}
```